widgetssdk/src/androidTest/java/com/glia/widgets/DownloadsFolderDataSourceTest.java

```diff
@@ -0,0 +1,59 @@
+package com.glia.widgets;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.glia.widgets.filepreview.data.source.local.DownloadsFolderDataSource;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+
+@RunWith(AndroidJUnit4.class)
+public class DownloadsFolderDataSourceTest {
+
+    private Context appContext;
+    private DownloadsFolderDataSource dataSource;
+
+    @Before
+    public void setUp() {
+        appContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        dataSource = new DownloadsFolderDataSource(appContext);
+    }
+
+    @Test
+    public void putImageToDownloads_returnsCompletableCompletes() {
+        dataSource.putImageToDownloads(IMAGE_NAME, BITMAP)
+                .test()
+                .assertComplete();
+    }
+
+    @Test
+    public void getImageFromDownloadsFolder_returnsMaybeOnError() {
```

Comment:
If the test method says "returnsMaybeOnError" I would expect to see some check that also validates that it returns Maybe. But even though it does return the Maybe it is not what this test checks so I think it can be changed to just "getImageFromDownloadsFolder_returnsError" which more clearly describes the test case. What is more, it would be nice to understand why we expect an error in this test.